src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs

```diff
@@ -760,27 +746,27 @@ private string[] GetFilterExtensions()
                 {
                     // Find our filter in the tokens list, then split it on the
                     // ';' character (which is the filter extension delimiter)
-                    string[] exts = tokens[indexOfExtension].Split(';');
+                    ReadOnlySpan<char> exts = tokens[indexOfExtension].AsSpan();
 
-                    foreach (string ext in exts)
+                    foreach (Range ext in exts.Split(';'))
                     {
                         // Filter extensions should be in the form *.txt or .txt,
                         // so we strip out everything before and including the '.'
                         // before adding the extension to our list.
                         // If the extension has no '.', we just ignore it as invalid.
-                        int i = ext.LastIndexOf('.');
+                        int i = exts[ext].LastIndexOf('.');
 
                         if (i >= 0)
                         {
-                            // start the substring one beyond the location of the '.'
+                            // start the slice one beyond the location of the '.'
                             // (i) and continue to the end of the string
-                            extensions.Add(ext.Substring(i + 1, ext.Length - (i + 1)));
+                            extensions.Add(exts[ext].Slice(i + 1, exts[ext].Length - (i + 1)).ToString());
```

Comment:
Isn't that equivalent to the more simple
```suggestion
                            extensions.Add(exts[ext].Slice(i + 1).ToString());
```
?

Reply:
When you think about it, it does. I didn't think about it, haha. Will address, thanks :)